Creates health score section in plugin management (#8036)
jenkins-infra/jenkins.io: content/doc/book/managing/plugins.adoc

The implied dependency can be removed by releasing a new version of the plugin that depends on a newer minimum Jenkins core. 
The link:/doc/developer/tutorial-improve/[Improve a plugin tutorial] provides steps that can assist plugin developers as they update the plugin to link:/doc/developer/tutorial-improve/update-base-jenkins-version/[depend on a newer minimum Jenkins core].

[[plugin-health-score]]
== What is the health score?

There is a metric available next to the plugins in the Plugin Manager called "Health Score".

image::managing/plugin-manager-available-health-score.png["Plugin Health Score in the Plugin Manager available listing", role=center]

You can find the same metric in the available update and the installed page of the plugin manager.

NOTE: The score value is always based on the latest available data.
This means that there is no score of a specific release of a plugin.
What is available on the installed and updates page of the plugin manager does not stand for the version already installed.

This metric is coming from link:https://github.com/jenkins-infra/plugin-health-scoring[Plugin Health Scoring].

This service is gathering data from each plugin.

> The data can be related to the source code of the plugin, the content of the plugin repository or external metrics about the plugin such as number of open pull requests, etc.

Reviewer: ```suggestion
The data can be related to the source code of the plugin, the content of the plugin repository, or external metrics about the plugin such as number of open pull requests.
```
Small suggestion to add oxford comma and remove use of `etc.` since there is only one example of the external metrics.

That data is then used to evaluate the status of the project.
All plugins are evaluated the same way, meaning there is a fair way to compare plugins.

> Only the score is shown, but the details behind this value are available on the link:https://plugins.jenkins.io[plugin site], in the health score section of each plugin:

Reviewer 2: ```suggestion
Only the score is shown, but the details behind this value are available on the link:https://plugins.jenkins.io[plugins site], in the health score section of each plugin:
```

> Only the score is shown, but the details behind this value are available on the link:https://plugins.jenkins.io[plugins site], in the health score section of each plugin:

Reviewer: ```suggestion
While only the score is shown, the details behind this value are available on the link:https://plugins.jenkins.io[plugins site] in the health score section of each plugin:
```
Small suggestion for syntax/grammar

> image::managing/plugin-site-health-score.png["Plugin health score on the plugin site", role=center]

Reviewer 2: ```suggestion
image::managing/plugin-site-health-score.png["Plugin health score on the plugins site", role=center]
```

It is also possible to find the details directly on the link:https://plugin-health.jenkins.io[Plugin Health Scoring] service.
To access it, you need to append the plugin name to the score URL.

> For example, the health score for `mailer` plugin is `https://plugin-health.jenkins.io/scores/mailer`.

Reviewer: ```suggestion
For example, the health score URL for the `Mailer` plugin is `https://plugin-health.jenkins.io/scores/mailer`.
```
Small suggestion for grammar/capitalization/syntax.

> Each section of the score can have one to multiple element evaluated.

Reviewer: ```suggestion
Each section of the score can have one or multiple elements evaluated.
```
Suggestion to adjust the grammar here so that elements matches multiple (plural).

> Each component of the section can have more or less impact on global score.

Reviewer: ```suggestion
Each component of the section can have more or less impact on the global score.
```
Small suggestion for syntax.

=== How to improve the scores?

The data gathered by the Plugin Health Scoring project on every plugin is evaluated frequently.

> Review the recommendations on the plugin site to improve the plugin score.

Reviewer 2: ```suggestion
Review the recommendations on the plugins site to improve the plugin score.
```

> Adopting a plugin, or contributing to each an every plugin is always welcome.

Reviewer: ```suggestion
Adopting a plugin or contributing to each and every plugin is always welcome.
```
Small adjustment for syntax.
